### ISSUE TITLE
Flaky test `02922_deduplication_with_zero_copy.sh` after 2024-06-18

### ISSUE THREAD

**Issue by fm4v** [ClickHouse/ClickHouse#65486]
Failure example: https://s3.amazonaws.com/clickhouse-test-reports/0/1bbc4bd2e2eaf642561d85a1cbb299a8ab708688/stateless_tests__release_.html

Matched with merge time of https://github.com/ClickHouse/ClickHouse/pull/65198

```
2024-06-18 11:37:13 Syntax error: failed at position 53 ('484e-80d9'):
2024-06-18 11:37:13 
2024-06-18 11:37:13 ls /clickhouse/zero_copy/zero_copy_s3/a6b3186e-deb7-[41;1m4[0m84e-80d9-ccd505300498
2024-06-18 11:37:13 
2024-06-18 11:37:13 Wrong number: '484e-80d9'
2024-06-18 11:37:13 , result:
2024-06-18 11:37:13 
2024-06-18 11:37:13 
2024-06-18 11:37:13 timeout expected, but loop exited with code: 0.
2024-06-18 11:37:13 the error is found if loop ends with 0.
2024-06-18 11:37:13 table_shared_id=a6b3186e-deb7-484e-80d9-ccd505300498
2024-06-18 11:37:13 
2024-06-18 11:37:13 stdout:
2024-06-18 11:37:13 
2024-06-18 11:37:13 timeout expected, but loop exited with code: 0.
2024-06-18 11:37:13 the error is found if loop ends with 0.
2024-06-18 11:37:13 table_shared_id=a6b3186e-deb7-484e-80d9-ccd505300498
2024-06-18 11:37:13 
2024-06-18 11:37:13 
2024-06-18 11:37:13 Settings used in the test: --max_insert_threads 0 --group_by_two_level_threshold 1000000 --group_by_two_level_threshold_bytes 50000000 --distributed_aggregation_memory_efficient 1 --fsync_metadata 1 --output_format_parallel_formatting 0 --input_format_parallel_parsing 1 --min_chunk_bytes_for_parallel_parsing 12709467 --max_read_buffer_size 924226 --prefer_localhost_replica 1 --max_block_size 72607 --max_joined_block_size_rows 69267 --max_threads 31 --optimize_append_index 1 --optimize_if_chain_to_multiif 0 --optimize_if_transform_strings_to_enum 1 --optimize_read_in_order 1 --optimize_or_like_chain 1 --optimize_substitute_columns 0 --enable_multiple_prewhere_read_steps 0 --read_in_order_two_level_merge_threshold 57 --optimize_aggregation_in_order 0 --aggregation_in_order_max_block_bytes 3755328 --use_uncompressed_cache 0 --min_bytes_to_use_direct_io 10737418240 --min_bytes_to_use_mmap_io 10737418240 --local_filesystem_read_method read --remote_filesystem_read_method read --local_filesystem_read_prefetch 0 --filesystem_cache_segments_batch_size 0 --read_from_filesystem_cache_if_exists_otherwise_bypass_cache 1 --throw_on_error_from_cache_on_write_operations 0 --remote_filesystem_read_prefetch 0 --allow_prefetched_read_pool_for_remote_filesystem 0 --filesystem_prefetch_max_memory_usage 128Mi --filesystem_prefetches_limit 0 --filesystem_prefetch_min_bytes_for_single_read_task 8Mi --filesystem_prefetch_step_marks 50 --filesystem_prefetch_step_bytes 100Mi --compile_aggregate_expressions 1 --compile_sort_description 1 --merge_tree_coarse_index_granularity 10 --optimize_distinct_in_order 0 --max_bytes_before_external_sort 10737418240 --max_bytes_before_external_group_by 0 --max_bytes_before_remerge_sort 2188867563 --min_compress_block_size 2437606 --max_compress_block_size 343611 --merge_tree_compact_parts_min_granules_to_multibuffer_read 61 --optimize_sorting_by_input_stream_properties 1 --http_response_buffer_size 416843 --http_wait_end_of_query True --enable_memory_bound_merging_of_aggregation_results 1 --min_count_to_compile_expression 3 --min_count_to_compile_aggregate_expression 3 --min_count_to_compile_sort_description 0 --session_timezone America/Argentina/Mendoza --prefer_warmed_unmerged_parts_seconds 4 --use_page_cache_for_disks_without_file_cache True --page_cache_inject_eviction False --merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability 0.51 --prefer_external_sort_block_bytes 100000000 --cross_join_min_rows_to_compress 0 --cross_join_min_bytes_to_compress 1 --min_external_table_block_size_bytes 0 --max_parsing_threads 1
2024-06-18 11:37:13 
2024-06-18 11:37:13 MergeTree settings used in test: --ratio_of_defaults_for_sparse_serialization 0.0 --prefer_fetch_merged_part_size_threshold 1 --vertical_merge_algorithm_min_rows_to_activate 1000000 --vertical_merge_algorithm_min_columns_to_activate 100 --allow_vertical_merges_from_compact_to_wide_parts 0 --min_merge_bytes_to_use_direct_io 5996275728 --index_granularity_bytes 28887177 --merge_max_block_size 21974 --index_granularity 59893 --min_bytes_for_wide_part 1073741824 --marks_compress_block_size 47147 --primary_key_compress_block_size 19852 --replace_long_file_name_to_hash 0 --m

```

**Comment by antonio2368** [comment]
Hmm based on the commit where it first fails it's actually this PR https://github.com/ClickHouse/ClickHouse/pull/61500
Somehow the highlighting ANSI code ended up in the Keeper path.

**Comment by alexey-milovidov** [comment]
@antonio2368, Obviously, ClickHouse SQL Lexer cannot be used for the Keeper client, because
`ls /clickhouse/zero_copy/zero_copy_s3/d6d39545-3f1f-43f6-994e-6bc8a7c1d662`
this is not SQL.

**Comment by alexey-milovidov** [comment]
My proposal is to only allow quoted string literals in Keeper client.

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse/ClickHouse/pull/65492

**Comment by antonio2368** [comment]
This fixes the test but leaves `keeper-client` broken.

**Comment by antonio2368** [comment]
Okay we can avoid the issue by also quoting the string, but in general, I would like to not have to use it.
So @pufit let's not use SQL parser for non-SQL stuff.

**Comment by alexey-milovidov** [comment]
Actually, it is ok to use our Lexer, if all command components are valid tokens.

For example, `/clickhouse/zero_copy/zero_copy_s3/d6d39545-3f1f-43f6-994e-6bc8a7c1d662` isn't a valid token, but `'/clickhouse/zero_copy/zero_copy_s3/d6d39545-3f1f-43f6-994e-6bc8a7c1d662'` is.

**Comment by alexey-milovidov** [comment]
@antonio2368 https://github.com/ClickHouse/ClickHouse/pull/65494

**Comment by Algunenano** [comment]
Whatever the decision, please make sure that the test is fixed in the 24.6 branch too
